src/design-system/ModalDialog.tsx

```diff
@@ -60,6 +60,11 @@ const ModalDialog: React.FC<Props> = (props) => {
   if (!open) return null;
 
   const handleOnClick = (event: React.MouseEvent<HTMLElement>) => {
+    // Stopping propogation so that the event does not fall through to elements
+    // "beneath" the modal.  This is useful in the case where we have a modal
+    // within a modal (i.e. a delete confirmation).
+    event.stopPropagation();
```

Comment:
Silly, but it took me forever to figure this out.  Without the `stopPropagation` in place, clicking anywhere within the top modal (including a form input) would cause the entire modal stack to collapse.